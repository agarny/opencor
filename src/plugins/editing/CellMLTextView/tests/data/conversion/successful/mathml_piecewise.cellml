<?xml version='1.0' encoding='UTF-8'?>
<model name="cellml_1_0" xmlns="http://www.cellml.org/cellml/1.0#" xmlns:cellml="http://www.cellml.org/cellml/1.0#">
    <component name="my_component">
        <math xmlns="http://www.w3.org/1998/Math/MathML">
            <apply>
                <eq/>
                <ci>a</ci>
                <piecewise>
                    <piece>
                        <cn cellml:units="dimensionless">123</cn>
                        <apply>
                            <eq/>
                            <ci>Version</ci>
                            <cn cellml:units="dimensionless">0</cn>
                        </apply>
                    </piece>
                </piecewise>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <piecewise>
                    <otherwise>
                        <cn cellml:units="dimensionless">123</cn>
                    </otherwise>
                </piecewise>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <piecewise>
                    <piece>
                        <cn cellml:units="dimensionless">123</cn>
                        <apply>
                            <eq/>
                            <ci>Version</ci>
                            <cn cellml:units="dimensionless">0</cn>
                        </apply>
                    </piece>
                    <piece>
                        <cn cellml:units="dimensionless">456</cn>
                        <apply>
                            <eq/>
                            <ci>Version</ci>
                            <cn cellml:units="dimensionless">1</cn>
                        </apply>
                    </piece>
                    <otherwise>
                        <cn cellml:units="dimensionless">789</cn>
                    </otherwise>
                </piecewise>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <piecewise>
                    <piece>
                        <cn cellml:units="dimensionless">123</cn>
                        <apply>
                            <eq/>
                            <ci>Version</ci>
                            <cn cellml:units="dimensionless">0</cn>
                        </apply>
                    </piece>
                    <otherwise>
                        <cn cellml:units="dimensionless">456</cn>
                    </otherwise>
                    <piece>
                        <cn cellml:units="dimensionless">789</cn>
                        <apply>
                            <eq/>
                            <ci>Version</ci>
                            <cn cellml:units="dimensionless">1</cn>
                        </apply>
                    </piece>
                </piecewise>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <piecewise>
                    <otherwise>
                        <cn cellml:units="dimensionless">123</cn>
                    </otherwise>
                    <piece>
                        <cn cellml:units="dimensionless">456</cn>
                        <apply>
                            <eq/>
                            <ci>Version</ci>
                            <cn cellml:units="dimensionless">0</cn>
                        </apply>
                    </piece>
                    <piece>
                        <cn cellml:units="dimensionless">789</cn>
                        <apply>
                            <eq/>
                            <ci>Version</ci>
                            <cn cellml:units="dimensionless">1</cn>
                        </apply>
                    </piece>
                </piecewise>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <ci>b</ci>
                    <piecewise>
                        <piece>
                            <cn cellml:units="dimensionless">123</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">0</cn>
                            </apply>
                        </piece>
                    </piecewise>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <ci>b</ci>
                    <piecewise>
                        <otherwise>
                            <cn cellml:units="dimensionless">123</cn>
                        </otherwise>
                    </piecewise>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <ci>b</ci>
                    <piecewise>
                        <piece>
                            <cn cellml:units="dimensionless">123</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">0</cn>
                            </apply>
                        </piece>
                        <piece>
                            <cn cellml:units="dimensionless">456</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">1</cn>
                            </apply>
                        </piece>
                        <otherwise>
                            <cn cellml:units="dimensionless">789</cn>
                        </otherwise>
                    </piecewise>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <ci>b</ci>
                    <piecewise>
                        <piece>
                            <cn cellml:units="dimensionless">123</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">0</cn>
                            </apply>
                        </piece>
                        <otherwise>
                            <cn cellml:units="dimensionless">456</cn>
                        </otherwise>
                        <piece>
                            <cn cellml:units="dimensionless">789</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">1</cn>
                            </apply>
                        </piece>
                    </piecewise>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <ci>b</ci>
                    <piecewise>
                        <otherwise>
                            <cn cellml:units="dimensionless">123</cn>
                        </otherwise>
                        <piece>
                            <cn cellml:units="dimensionless">456</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">0</cn>
                            </apply>
                        </piece>
                        <piece>
                            <cn cellml:units="dimensionless">789</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">1</cn>
                            </apply>
                        </piece>
                    </piecewise>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <ci>b</ci>
                    <piecewise>
                        <piece>
                            <cn cellml:units="dimensionless">123</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">0</cn>
                            </apply>
                        </piece>
                    </piecewise>
                    <ci>b</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <ci>b</ci>
                    <piecewise>
                        <otherwise>
                            <cn cellml:units="dimensionless">123</cn>
                        </otherwise>
                    </piecewise>
                    <ci>b</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <ci>b</ci>
                    <piecewise>
                        <piece>
                            <cn cellml:units="dimensionless">123</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">0</cn>
                            </apply>
                        </piece>
                        <piece>
                            <cn cellml:units="dimensionless">456</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">1</cn>
                            </apply>
                        </piece>
                        <otherwise>
                            <cn cellml:units="dimensionless">789</cn>
                        </otherwise>
                    </piecewise>
                    <ci>b</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <ci>b</ci>
                    <piecewise>
                        <piece>
                            <cn cellml:units="dimensionless">123</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">0</cn>
                            </apply>
                        </piece>
                        <otherwise>
                            <cn cellml:units="dimensionless">456</cn>
                        </otherwise>
                        <piece>
                            <cn cellml:units="dimensionless">789</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">1</cn>
                            </apply>
                        </piece>
                    </piecewise>
                    <ci>b</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <ci>b</ci>
                    <piecewise>
                        <otherwise>
                            <cn cellml:units="dimensionless">123</cn>
                        </otherwise>
                        <piece>
                            <cn cellml:units="dimensionless">456</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">0</cn>
                            </apply>
                        </piece>
                        <piece>
                            <cn cellml:units="dimensionless">789</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">1</cn>
                            </apply>
                        </piece>
                    </piecewise>
                    <ci>b</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <piecewise>
                        <piece>
                            <cn cellml:units="dimensionless">123</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">0</cn>
                            </apply>
                        </piece>
                    </piecewise>
                    <ci>b</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <piecewise>
                        <otherwise>
                            <cn cellml:units="dimensionless">123</cn>
                        </otherwise>
                    </piecewise>
                    <ci>b</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <piecewise>
                        <piece>
                            <cn cellml:units="dimensionless">123</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">0</cn>
                            </apply>
                        </piece>
                        <piece>
                            <cn cellml:units="dimensionless">456</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">1</cn>
                            </apply>
                        </piece>
                        <otherwise>
                            <cn cellml:units="dimensionless">789</cn>
                        </otherwise>
                    </piecewise>
                    <ci>b</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <piecewise>
                        <piece>
                            <cn cellml:units="dimensionless">123</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">0</cn>
                            </apply>
                        </piece>
                        <otherwise>
                            <cn cellml:units="dimensionless">456</cn>
                        </otherwise>
                        <piece>
                            <cn cellml:units="dimensionless">789</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">1</cn>
                            </apply>
                        </piece>
                    </piecewise>
                    <ci>b</ci>
                </apply>
            </apply>
            <apply>
                <eq/>
                <ci>a</ci>
                <apply>
                    <plus/>
                    <piecewise>
                        <otherwise>
                            <cn cellml:units="dimensionless">123</cn>
                        </otherwise>
                        <piece>
                            <cn cellml:units="dimensionless">456</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">0</cn>
                            </apply>
                        </piece>
                        <piece>
                            <cn cellml:units="dimensionless">789</cn>
                            <apply>
                                <eq/>
                                <ci>Version</ci>
                                <cn cellml:units="dimensionless">1</cn>
                            </apply>
                        </piece>
                    </piecewise>
                    <ci>b</ci>
                </apply>
            </apply>
        </math>
    </component>
</model>
