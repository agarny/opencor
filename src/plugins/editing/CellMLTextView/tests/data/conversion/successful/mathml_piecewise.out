def model cellml_1_0 as
    def comp my_component as
        a = sel
            case Version == 0{dimensionless}:
                123{dimensionless};
        endsel;

        a = sel
            otherwise:
                123{dimensionless};
        endsel;

        a = sel
            case Version == 0{dimensionless}:
                123{dimensionless};
            case Version == 1{dimensionless}:
                456{dimensionless};
            otherwise:
                789{dimensionless};
        endsel;

        a = sel
            case Version == 0{dimensionless}:
                123{dimensionless};
            otherwise:
                456{dimensionless};
            case Version == 1{dimensionless}:
                789{dimensionless};
        endsel;

        a = sel
            otherwise:
                123{dimensionless};
            case Version == 0{dimensionless}:
                456{dimensionless};
            case Version == 1{dimensionless}:
                789{dimensionless};
        endsel;

        a = b+sel(case Version == 0{dimensionless}: 123{dimensionless});
        a = b+sel(otherwise: 123{dimensionless});
        a = b+sel(case Version == 0{dimensionless}: 123{dimensionless}, case Version == 1{dimensionless}: 456{dimensionless}, otherwise: 789{dimensionless});
        a = b+sel(case Version == 0{dimensionless}: 123{dimensionless}, otherwise: 456{dimensionless}, case Version == 1{dimensionless}: 789{dimensionless});
        a = b+sel(otherwise: 123{dimensionless}, case Version == 0{dimensionless}: 456{dimensionless}, case Version == 1{dimensionless}: 789{dimensionless});
        a = b+sel(case Version == 0{dimensionless}: 123{dimensionless})+b;
        a = b+sel(otherwise: 123{dimensionless})+b;
        a = b+sel(case Version == 0{dimensionless}: 123{dimensionless}, case Version == 1{dimensionless}: 456{dimensionless}, otherwise: 789{dimensionless})+b;
        a = b+sel(case Version == 0{dimensionless}: 123{dimensionless}, otherwise: 456{dimensionless}, case Version == 1{dimensionless}: 789{dimensionless})+b;
        a = b+sel(otherwise: 123{dimensionless}, case Version == 0{dimensionless}: 456{dimensionless}, case Version == 1{dimensionless}: 789{dimensionless})+b;
        a = sel(case Version == 0{dimensionless}: 123{dimensionless})+b;
        a = sel(otherwise: 123{dimensionless})+b;
        a = sel(case Version == 0{dimensionless}: 123{dimensionless}, case Version == 1{dimensionless}: 456{dimensionless}, otherwise: 789{dimensionless})+b;
        a = sel(case Version == 0{dimensionless}: 123{dimensionless}, otherwise: 456{dimensionless}, case Version == 1{dimensionless}: 789{dimensionless})+b;
        a = sel(otherwise: 123{dimensionless}, case Version == 0{dimensionless}: 456{dimensionless}, case Version == 1{dimensionless}: 789{dimensionless})+b;
    enddef;
enddef;
